Lucas

Geotricity opened the pull request:
Constants class:
- Added Controller Port number
- Added constants to control robot movement

Driver Class:
- Imported commands
- Added Dualshock controller
-  Added methods to get throttle and steer

Geotricity committed:
Gives basic functionality

- Imported commands
- Added Dualshock controller
-  Added methods to get throttle and steer

Geotricity committed:
Update Constants.java

- Added Controller Port number
- Added constants to control robot movement

Geotricity committed:
Oops

get rid of these

danielzyy commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,27 @@
+
+
+    // SETTINGS
+    public static final double DEADZONE = 0.15; // Deadzone applied to joysticks to aid in adjusting sensitivity
```
Is this deadzone applied directly to the raw controller values and not the speed values(like from `getY()` and `getX()`)?  If not, and it is meant to be applied on the actual motor speed, I would be wondering why the deadzone is half of the `MAX_THROTTLE_SPEED`, since that seems pretty large of a deadzone.

Geotricity replied:
You're right, it is a pretty large deadzone. I've changed the code so that it is applied on the raw controller values.

bricktrooper commented on src/main/java/frc/robot/controls/Driver.java:
```diff
@@ -41,4 +46,48 @@ public Driver(Robot robot)
+    ===================================== */
+    public double getThrottle()
+    {
+        double throttleValue = 0;
+        
+        throttleValue = -dualshock.getY();
+        throttleValue *= Constants.MAX_THROTTLE_SPEED;
```
If you want, you could do these all in one line:
```
double throttleValue = -dualshock.getY() * Constants.MAX_THROTTLE_SPEED;
```
I think it might run a little faster unless the Java compiler is super smart :P

Geotricity replied:
I'm not able to do this all on one line since the deadzone check has been moved to between reading the value from the joystick and scaling the input.

bricktrooper commented on src/main/java/frc/robot/controls/Driver.java:
```diff
@@ -41,4 +46,48 @@ public Driver(Robot robot)
+    ===================================== */
+    public double getSteer()
+    {
+        double steerValue = 0;
+
+        steerValue = dualshock.getZ();
+        steerValue *= Constants.MAX_STEER_SPEED;
```
Same comment as above

danielzyy replied:
set the `steerValue` directly to `dualshock.getZ() * Constants.MAX_STEER_SPEED` in one line instead of 2 for clarity.

danielzyy commented on src/main/java/frc/robot/controls/Driver.java:
```diff
@@ -41,4 +46,48 @@ public Driver(Robot robot)
+        throttleValue = -dualshock.getY();
+        throttleValue *= Constants.MAX_THROTTLE_SPEED;
+
+        if (Math.abs(throttleValue) < Constants.DEADZONE)
```
Be careful here, you've already updated `throttleValue` to be a fraction of `MAX_THROTTLE_SPEED` which I am assuming is the speed you are setting to the motors, but you are comparing it to the controller deadzone? If that `DEADZONE` is referring to the raw controller value from `dualshock.getY()`, you should be checking that in a separate if-statement before line 61 to immediately return if controller reading is in deadzone Ex `if(Math.abs(dualshock.getY()) < Constants.DEADZONE) return 0;`

Otherwise, right now the robot will not move until you hit a throttleValue of more than 0.15

bricktrooper replied:
This is a good point!  Maybe we should be checking the deadzone before applying the scaling factors.

bricktrooper commented on src/main/java/frc/robot/controls/Driver.java:
```diff
@@ -13,12 +13,16 @@
 {
     // JOYSTICKS / CONTROLLERS //
-
+    private Joystick dualshock;
```
Is dualshock a joystick or an Xbox/playstation controller?  You may want to ask @droid-101 what kind of controller he will be driving with.  If dualshock is used by the operator then you may want to move this code to `Operator.java`.

Geotricity replied:
The dualshock is a generic playstation controller. I'm not yet sure what kind of controller @droid-101 wants to use, but I can always move the code over in a later pull request.

bricktrooper replied:
Sounds good to me!

Geotricity replied:
Ok, I heard from @droid-101 that he wants 2 joysticks for the driver and a controller for the operator. Should I change that here or do that in a different pull request?

danielzyy commented on src/main/java/frc/robot/controls/Driver.java:
```diff
@@ -41,4 +46,48 @@ public Driver(Robot robot)
+        steerValue = dualshock.getZ();
+        steerValue *= Constants.MAX_STEER_SPEED;
+
+        if (Math.abs(steerValue) < Constants.DEADZONE)
```
Same thing here check previous comment for line 64, are you checking _controller deadzone_ or actual _motor deadzone_?

bricktrooper commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,27 @@
+    // HARDWARE PORTS
+   
+
+    // CONTROLLER COMPUTER PORTS
+    public static final int DUALSHOCK = 0;
+
+    // LIMIT SWITCH PORTS
+
+
+    // SETTINGS
+    public static final double DEADZONE = 0.15; // Deadzone applied to joysticks to aid in adjusting sensitivity
+    public static final double MAX_THROTTLE_SPEED = 0.30;
+    public static final double MAX_STEER_SPEED = 0.2;
```
You can actually just put these constants inside `Driver.java` and `Operator.java` right at the top of the classes.  This year we decided to get rid of `Constants.java` and sort the constants into which ever file they are used in.  Keeps it more organized this way :)

danielzyy commented on src/main/java/frc/robot/controls/Driver.java:
```diff
@@ -41,4 +46,48 @@ public Driver(Robot robot)
+        double throttleValue = 0;
+        
+        throttleValue = -dualshock.getY();
+        throttleValue *= Constants.MAX_THROTTLE_SPEED;
```
set the `throttleValue` directly to `-dualshock.getY() * Constants.MAX_THROTTLE_SPEED` in one line instead of 2 for clarity.

danielzyy reviewed (changes requested):
Review for controller deadzone and steerValue and throttleValue deadzone condition.

Geotricity committed:
Add intake functionality

- Added methods in intake subsystem
- Added initialized method for the intake in robot
- Added intake constants

bricktrooper reviewed (changes requested):
This looks pretty good so far, just some small things to consider.  Also it seems like your code is failing to compile.  Try building it on your VSCode and see if you can find out what's wrong.  You can also check the "details" next to the build checks and see the compile log.

bricktrooper commented:
Also just a tip for the future, if you are also working on the subsystems such as `Intake.java`, best to keep it in a separate PR just to keep things organized.

Geotricity committed:
Fixed Requested Changes to #16

 - Removed Constants.java and placed constants in the classes where they are used
- Changed DEADZONE check to occur before the joystick input is scaled
- Commented out a line in ExampleCommand that seemed to be causing the compilation error

Geotricity commented:
I've made the changes requested here in the newest commit by:

- Removing Constants.java and placed constants in the classes where they are used
- Changing the DEADZONE check to occur before the joystick input is scaled
- Commenting out a line in ExampleCommand that seemed to be causing the compilation error

Geotricity commented:
> Also just a tip for the future, if you are also working on the subsystems such as `Intake.java`, best to keep it in a separate PR just to keep things organized.

Thanks, I'll do that next time. :)

bricktrooper reviewed (approved):
Looks good man, I think you can merge this for now unless anyone else has any comments.  Also make sure to change update your PR title when merging and just replace the auto-generated description with the nice bullet points you wrote at the top.  btw you might have to hit "resolve" on all the comments before it lets you merge.

danielzyy reviewed (approved):
Looks good to me
